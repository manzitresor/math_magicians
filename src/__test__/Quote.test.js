import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../components/quotes';

describe('Quote render', () => {
  test('should render loading state correctly', () => {
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render error state correctly', () => {
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render quote correctly', () => {
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render loading state correctly', () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve([]),
    }));
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
    global.fetch.mockRestore();
  });

  test('should render error state correctly', () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.reject(new Error('API Error')));
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
    global.fetch.mockRestore();
  });

  test('should render data state correctly', async () => {
    const fakeData = {
      quote: 'test quote',
    };
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(fakeData),
    }));
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
    global.fetch.mockRestore();
  });
});
