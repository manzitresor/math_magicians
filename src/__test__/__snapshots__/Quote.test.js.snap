// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote render should render data state correctly 1`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Quote render should render error state correctly 1`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Quote render should render error state correctly 2`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Quote render should render loading state correctly 1`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Quote render should render loading state correctly 2`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Quote render should render quote correctly 1`] = `
<div
  class="loading"
>
  <p>
    Loading...
  </p>
</div>
`;
